appgate/syncer/operator.py

```diff
@@ -0,0 +1,285 @@
+import asyncio
+from asyncio import Queue
+import os
+from pathlib import Path
+import sys
+from typing import List, Optional, Callable, Dict, Tuple
+
+from kubernetes.config import ConfigException
+
+from appgate.logger import log
+from appgate.openapi.openapi import SPEC_DIR, generate_api_spec
+from appgate.operator import init_kubernetes, run_k8s
+from appgate.secrets import k8s_get_secret
+from appgate.types import (
+    AppgateEvent,
+    AppgateEventError,
+    EntityWrapper,
+    GitOperatorContext,
+    GitOperatorArguments,
+    NAMESPACE_ENV,
+    SPEC_DIR_ENV,
+    APPGATE_SECRETS_KEY,
+    TIMEOUT_ENV,
+    get_tags,
+    APPGATE_TARGET_TAGS_ENV,
+    get_dry_run,
+    ensure_env,
+    GIT_REPOSITORY_ENV,
+    GIT_VENDOR_ENV,
+    GIT_BASE_BRANCH_ENV,
+    GIT_DUMP_DIR,
+    APPGATE_LOG_LEVEL,
+    GIT_REPOSITORY_FORK_ENV,
+    BUILTIN_TAGS,
+    EntityClient,
+    is_target,
+    GitCommitState,
+    GIT_REPOSITORY_MAIN_BRANCH_ENV,
+    GIT_REPOSITORY_MAIN_BRANCH,
+)
+from appgate.openapi.types import (
+    AppgateException,
+    APISpec,
+    MissingFieldDependencies,
+)
+from appgate.state import (
+    appgate_state_empty,
+    create_appgate_plan,
+    appgate_plan_apply,
+    resolve_appgate_state,
+    entities_conflict_summary,
+)
+from appgate.syncer.git import (
+    get_git_repository,
+    GitRepo,
+    get_current_branch_state,
+    GitEntityClient,
+)
+
+
+def git_operator_context(
+    args: GitOperatorArguments,
+    k8s_get_secret: Optional[Callable[[str, str], str]] = None,
+    namespace: str | None = None,
+) -> GitOperatorContext:
+    namespace = namespace or args.namespace or os.getenv(NAMESPACE_ENV)
+    spec_directory = os.getenv(SPEC_DIR_ENV) or args.spec_directory or SPEC_DIR
+    secrets_key = os.getenv(APPGATE_SECRETS_KEY)
+    api_spec = generate_api_spec(
+        spec_directory=Path(spec_directory) if spec_directory else None,
+        secrets_key=secrets_key,
+        k8s_get_secret=k8s_get_secret,
+    )
+    if not namespace:
+        raise AppgateException(
+            "Namespace must be defined in order to run the git-operator"
+        )
+    dry_run_mode = get_dry_run(args.no_dry_run)
+    return GitOperatorContext(
+        namespace=namespace,
+        api_spec=api_spec,
+        timeout=int(os.getenv(TIMEOUT_ENV) or args.timeout),
+        target_tags=get_tags(args.target_tags, os.getenv(APPGATE_TARGET_TAGS_ENV)),
+        dry_run=dry_run_mode,
+        git_vendor=ensure_env(GIT_VENDOR_ENV),
+        git_repository=ensure_env(GIT_REPOSITORY_ENV),
+        git_base_branch=ensure_env(GIT_BASE_BRANCH_ENV),
+        log_level=os.environ.get(APPGATE_LOG_LEVEL, "info"),
+        git_repository_fork=os.environ.get(GIT_REPOSITORY_FORK_ENV),
+        main_branch=os.environ.get(
+            GIT_REPOSITORY_MAIN_BRANCH_ENV, GIT_REPOSITORY_MAIN_BRANCH
+        ),
+    )
```

Comment:
What is the difference between `git_base_branch` and `main_branch`?

Reply:
None, I think that's just a duplication. I will remove one.